react/week-2/lesson.md

```diff
@@ -429,6 +440,66 @@ To help us cleanly split up code that performs business logic from code that sho
 
 Container components usually have some state and handler methods. Because of this they must use the `class` syntax. Presentational components on the other hand don't require the more verbose syntax. Instead they usually use the functional syntax.
 
+## React Hooks
+
+React is continously updated with features all the time, one that stands out the most are **React Hooks**, released in React v16.8.
+
+### What are hooks?
+
+**Hooks** are functions that enable you manipulate the React state (and other lifecycle features, which you will be learning next week).
+
+### Why React hooks, over Class component state and methods?
+
+- Code is less complex - being able to make your code more readable in the long term, make it more maintainable
+- Hooks are reusable - unlike methods that scoped within in a class components, you are able to create your own and reuse them elsewhere you in the code. Keeping your code DRY (Don't Repeat Yourself)
+
+
+### Counter Example - with hooks
+
+```js
+import React, { useState } from 'react';
+
+function Counter() {
+  // Declare a new state variable, which we'll call "count"
+  const [count, setCount] = useState(0);
+
+  return (
+    <div>
+      <p>You clicked {count} times</p>
+      <button onClick={() => setCount(count + 1)}>
+        Click me
+      </button>
+    </div>
+  );
+}
+```
+
+Here, `useState` is a Hook. We call it inside a function component to add some local state to it. React will preserve this state between re-renders. `useState` returns a pair: the current state i.e. `count` and a function that lets you update it i.e. `setCount`.
+
+Note that variable `count` and function `setCount` come from destructured arrays (a JavaScript feature). The initial value of count is `0` and whenever we press on the button, it will fire the setCount function to increment by 1.
+
+### Setting multiple states
+
+```js
+function ExampleWithManyStates() {
+  // Declare multiple state variables!
+  const [age, setAge] = useState(0);
+  const [fruit, setFruit] = useState('Oranges');
+  
+  return (
+    <div>
+      <p> Hi! I am Alex </p>
+      <p> {`I am ${age} years old`} </p>
+      <p> {`I prefer to eat ${fruit}`} </p>      
```

Comment:
Sorry for only just noticing this, but I would tend to write this like:

```suggestion
      <p> I am {age} years old </p>
      <p> I prefer to eat {fruit} </p> 
```

Was there a reason to use template strings?